services/construction_service.go

```diff
@@ -1048,10 +1049,44 @@ func encodeMethodArgsStrings(sigData []byte, methodSig string, methodArgs []stri
 				argData = value
 			}
 		case strings.HasPrefix(v, "bytes"):
-			{
-				var value []byte
-				copy(value, methodArgs[i])
+			if v == "bytes" {
+				// Converts dynamically-sized byte array to a slice
+				value, err := hexutil.Decode(methodArgs[i])
+				if err != nil {
+					return nil, err
+				}
 				argData = value
+			} else {
+				// Converts fixed-size byte array (like bytes32) to array
+				sizeStr := strings.TrimPrefix(v, "bytes")
+				size, err := strconv.Atoi(sizeStr)
+				if err != nil {
+					log.Fatal(err)
```

Comment:
Since we're handling user input, should return an error here rather than panic.

Reply:
fixed!